playground/AspireWithNode/AspireWithNode.AppHost/AppHost.cs

```diff
@@ -23,7 +26,19 @@
 
 if (builder.Environment.IsDevelopment() && launchProfile == "https")
 {
-    frontend.RunWithHttpsDevCertificate("HTTPS_CERT_FILE", "HTTPS_CERT_KEY_FILE");
+    frontend
+        .WithHttpsEndpoint(env: "HTTPS_PORT")
+        .WithCertificateKeyPairConfiguration(ctx =>
+        {
+            ctx.EnvironmentVariables["HTTPS_CERT_FILE"] = ctx.CertificatePath;
+            ctx.EnvironmentVariables["HTTPS_CERT_KEY_FILE"] = ctx.KeyPath;
+            return Task.CompletedTask;
+        })
+        .WithEnvironment(ctx =>
+        {
+            var httpsEndpoint = frontend.GetEndpoint("https");
+            ctx.EnvironmentVariables["HTTPS_REDIRECT_PORT"] = ReferenceExpression.Create($"{httpsEndpoint.Property(EndpointProperty.Port)}");
```

Comment:
Does not need to be a reference expression.

```suggestion
            ctx.EnvironmentVariables["HTTPS_REDIRECT_PORT"] = httpsEndpoint.Property(EndpointProperty.Port);
```